pwndbg/commands/xinfo.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import gdb
+import subprocess
+
+import pwndbg.arch
+import pwndbg.commands
+import pwndbg.config
+import pwndbg.memory
+import pwndbg.vmmap
+import pwndbg.regs
+import pwndbg.color.memory as M
+
+
+parser = argparse.ArgumentParser(description='Shows offsets of the specified address to useful other locations')
+parser.add_argument('address', nargs='?', default='$pc', 
+                    help='Address to inspect')
+
+def print_line(name, addr, first, second, op, width = 20):
+    FMT = "{} {} = {} {} {:#x}"
+
+    print(FMT.format(
+        name.rjust(width),
+        M.get(addr),
+        M.get(first) if type(first) is not str else first.ljust(len(hex(addr))),
+        op,
+        second,))
+
+@pwndbg.commands.ArgparsedCommand(parser)
+@pwndbg.commands.OnlyWhenRunning
+def xinfo(address=None):
+    addr = int(address)
+    addr &= pwndbg.arch.ptrmask
+
+    page = pwndbg.vmmap.find(addr)
+
+    print("Extended information for virtual address {}:".format(M.get(addr)))
+
+    if page is None:
+        print("\n  Virtual address {:#x} is not mapped.".format(addr))
+        return
+
+    print("\n  Containing mapping:")
+    print(M.get(address, text=str(page)))
+
+    print("\n  Offset information:")
+
+    if page.objfile == "[stack]":
+        # If it's a stack address, print offsets to top and bottom of stack, as
+        # well as offsets to current stack and base pointer (if used by debugee)
+        sp = pwndbg.regs.sp
+        frame = pwndbg.regs[pwndbg.regs.frame]
+        frame_mapping = pwndbg.vmmap.find(frame)
+
+        print_line("Stack Top", addr, page.vaddr, addr - page.vaddr, "+")
```

Comment:
We could potentially use https://github.com/pwndbg/pwndbg/blob/ed887fdbc48c27019211d71794f40b94ba94acb5/pwndbg/memory.py#L387-L400 however since we also subtract stuff here, maybe it isn't the best idea.
I leave it up to you.